src/transaction.rs

```diff
@@ -470,29 +470,15 @@ impl Transaction {
     }
 
     /// Prepare the transaction for VM initialization for script execution
-    ///
-    /// The callback argument is to expect a contract id and return its balance root and state root
     #[cfg(feature = "std")]
-    pub fn prepare_init_script<F>(&mut self, mut f: F) -> io::Result<&mut Self>
-    where
-        F: FnMut(&fuel_types::ContractId) -> io::Result<(Bytes32, Bytes32)>,
-    {
-        let id = self.id();
-
-        let (inputs, outputs) = match self {
-            Transaction::Script {
-                inputs, outputs, ..
-            }
-            | Transaction::Create {
-                inputs, outputs, ..
-            } => (inputs, outputs),
+    pub fn prepare_init_script(&mut self) -> io::Result<&mut Self> {
```

Comment:
nit: Add a comment that the tx is expected to be sent to the VM with the balance/state root set by the client in the inputs